src/extra/helpers/helpers.cpp

```diff
@@ -0,0 +1,139 @@
+
+    if (_contextVarRef) {
+      if (_contextVarRef->refcount > 1) {
+        SHLOG_ERROR("Found {} dangling reference(s) to {}", _contextVarRef->refcount - 1, Context::contextVarName);
+      }
+      releaseVariable(_contextVarRef);
+    }
+  }
+
+  SHTypeInfo compose(SHInstanceData &data) {
+    composeCheckMainThread(data);
+    composeCheckMainWindowGlobals(data);
+
+    std::vector<SHExposedTypeInfo> exposed(PtrIterator(data.shared.elements),
+                                           PtrIterator(data.shared.elements + data.shared.len));
+    exposed.push_back(ExposedInfo::ProtectedVariable(Context::contextVarName, SHCCSTR("Helper context"), Context::Type));
+
+    if (!_queue.isVariable())
+      throw ComposeError("Queue not set");
+
+    if (!_view.isVariable())
+      throw ComposeError("View not set");
+
+    SHInstanceData contentInstanceData = data;
+    contentInstanceData.shared.elements = exposed.data();
+    contentInstanceData.shared.len = exposed.size();
+    return _content.compose(contentInstanceData).outputType;
+  }
+
+  void handleGizmoInputEvents(std::vector<SDL_Event> &events) {
+    for (auto &event : events) {
+      if (event.type == SDL_MOUSEMOTION) {
+        _cursorPosition.x = event.motion.x;
+        _cursorPosition.y = event.motion.y;
+      } else if (event.type == SDL_MOUSEBUTTONDOWN || event.type == SDL_MOUSEBUTTONUP) {
+        _cursorPosition.x = event.button.x;
+        _cursorPosition.y = event.button.y;
+        if (event.button.state == SDL_PRESSED)
+          _mouseButtonState |= SDL_BUTTON(event.button.button);
+        else
+          _mouseButtonState &= ~SDL_BUTTON(event.button.button);
+      }
+    }
+  }
+
+  SHVar activate(SHContext *shContext, const SHVar &input) {
+    Var queueVar(_queue.get());
+    Var viewVar(_view.get());
+
+    ViewPtr view = static_cast<SHView *>(viewVar.payload.objectValue)->view;
+
+    assert(queueVar.payload.objectValue);
+    _context.queue = static_cast<SHDrawQueue *>(queueVar.payload.objectValue)->queue;
+    assert(_context.queue);
+
+    gfx::Context &gfxContext = getContext();
+    gfx::MainWindowGlobals &mainWindowGlobals = getMainWindowGlobals();
+    int2 outputSize = gfxContext.getMainOutputSize();
+
+    handleGizmoInputEvents(mainWindowGlobals.events);
+
+    gfx::gizmos::InputState gizmoInput;
+    gizmoInput.cursorPosition = _cursorPosition;
+    gizmoInput.pressed = _mouseButtonState;
+    gizmoInput.viewSize = outputSize;
+
+    // need to update the context var
+    _contextVarRef->payload.objectValue = &_context;
+
+    _context.gizmoContext.begin(gizmoInput, view);
+
+    SHVar _shardsOutput{};
+    _content.activate(shContext, input, _shardsOutput);
+
+    _context.gizmoContext.end(_context.queue);
+
+    return _shardsOutput;
+  }
+};
+
+extern void registerHighlightShards();
+extern void registerGizmoShards();
+extern void registerShapeShards();
+void registerShards() {
+  REGISTER_SHARD("Helpers.Context", HelperContextShard);
```

Comment:
Not sure on this, anyway the naming is confusing.

Reply:
I think we should do `Helpers` -> `Gizmos`, they are all gizmos by definition. An "unnecessary" tool (when the product is shipped).
This allows you to remove the `Gizmo` suffix you used at times.